fuel/datasets/hdf5.py

```diff
@@ -316,11 +322,21 @@ def _out_of_memory_get_data(self, state=None, request=None):
             if isinstance(request, slice):
                 request = slice(request.start + subset.start,
                                 request.stop + subset.start, request.step)
+                data = state[source_name][request]
             elif isinstance(request, list):
                 request = [index + subset.start for index in request]
+                try:
+                    data = state[source_name][request]
+                except TypeError:
```

Comment:
Please document that prominently, both in a comment adjacent to the code
and in relevant class/method docstrings. That's the sort of "whoopsie"
feature that can catch someone off guard.

>